torchtitan/distributed/utils.py

```diff
@@ -303,5 +303,6 @@ def clip_grad_norm_(
             dist.all_reduce(total_norm, op=dist.ReduceOp.SUM, group=pp_mesh.get_group())
             total_norm **= 1.0 / norm_type
 
-    torch.nn.utils.clip_grads_with_norm_(parameters, max_norm, total_norm, foreach)
+    if max_norm > 0:
```

Comment:
I'd rather make a flag in train.py clip/not clip, so that we don't spend time on norm computation and dist. communication